services/jupyter/src/notebooks/rubrix/rubrix-active-learning-loop.ipynb

```diff
@@ -0,0 +1,701 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 120,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# validation dataset\n",
+    "val_text = [row[\"text\"] for row in labeled_dataset[\"validation\"]][:20]\n",
+    "val_labels = [row[\"label\"] for row in labeled_dataset[\"validation\"]][:20]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 121,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "val_dataset = TransformersDataset.from_arrays(\n",
+    "    val_text,\n",
+    "    val_labels,\n",
+    "    tokenizer=TOKENIZER,\n",
+    "    target_labels=[0, 1, 2, 3]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    ">Notes\n",
+    "\n",
+    "- cleaning??\n",
+    "- test set???"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Set up active learner\n",
+    "\n",
+    "- Component 1 -> classifier\n",
+    "- Component 2 -> query strategy (sampling)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 122,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# hyper-params\n",
+    "params = dict(\n",
+    "        {\n",
+    "            \"lr\": 3e-5,\n",
+    "            \"num_epochs\": 1,\n",
+    "            \"mini_batch_size\": 32,\n",
+    "            \"model_selection\": True,\n",
+    "            \"device\": \"cuda\"\n",
+    "        \n",
+    "    }\n",
+    ")\n",
+    "\n",
+    "# classifier\n",
+    "model_factory = TransformerBasedClassificationFactory(\n",
+    "    TransformerModelArguments(TRANSFORMER_MODEL),\n",
+    "    num_classes=NUM_CLASSES,\n",
+    "    kwargs=params\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 123,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# define query strategy\n",
+    "query_strategy = BreakingTies()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 124,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def initialize_active_learner(active_learner, train_labels, weak_train_labels):\n",
+    "\n",
+    "    indices_initial = [i for i in range(len(weak_train_labels))]\n",
+    "    active_learner.initialize_data(indices_initial, train_labels[indices_initial])\n",
+    "\n",
+    "    return indices_initial"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 129,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "active_learner = PoolBasedActiveLearner(\n",
+    "    model_factory,\n",
+    "    query_strategy,\n",
+    "    dataset,\n",
+    "    reuse_model=True\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 130,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# this takes quite a bit of time... (1hr!!) - we initilize the active learner with the weak labels\n",
+    "indices_labeled = initialize_active_learner(active_learner, train_labels, weak_train_labels)"
```

Comment:
If I understood the documentation correctly, this line fine-tunes the model with the weak labels, right?

Reply:
yeah exactly